AutomatedTesting/Gem/PythonTests/Atom/tests/MaterialEditor_Atom_ExpectsTestTimeout.py

```diff
@@ -28,22 +28,27 @@ def MaterialEditor_TestsTimeout_TimeoutFailure():
 
     :return: None
     """
-    import Atom.atom_utils.atom_tools_utils as atom_tools_utils
+    import time
 
+    import Atom.atom_utils.atom_tools_utils as atom_tools_utils
     from editor_python_test_tools.utils import Report, Tracer, TestHelper
 
     with Tracer() as error_tracer:
+        # 1. Create start_time variable for the callback function to compare against.
+        start_time = time.time() + 12
 
-        # 1. Create callback function for timing out the test.
+        # 2. Create callback function for timing out the test.
         def loop_variable():
+            if time.time() > start_time:
+                return False
             return True
 
-        # 2. Trigger test timeout.
+        # 3. Wait for test timeout from test launcher script before this Report can pass (xfail expected).
```

Comment:
looking at the whole PR the pytest level is setting a timeout of 10 and this Report.result has a timeout of 15 with a loop_variable that is finishing in 12. so the expected behavior is that at the pytest/multitest framework level the test should be timed out before reaching this Report.result.
is it explicitly clear in these comments that you expect the framework (pytest/multitest) to timeout and xfail this test before it reaches the actual Report.result conclusion?
I understand what's going on but I feel less clear the more I spend time reviewing this code.

Reply:
i added the delay block before a `Report.success()` call now, so it should be more clear.

unfortunately `idle_wait()` nor `idle_wait_frames()` appear available for MaterialEditor tests, so I stuck with my `time` loop.